velox/common/caching/StringIdMap.cpp

```diff
@@ -43,13 +43,126 @@ void StringIdMap::release(uint64_t id) {
   }
 }
 
+#define VELOX_STRINGIDMAP_PREFIX "[StringIdMap] "
+#define VELOX_STRINGIDMAP_ERROR_LOG LOG(ERROR) << VELOX_STRINGIDMAP_PREFIX
+
+void StringIdMap::checkAndReportIntegrity(uint64_t id) const {
+  if (idToString_.size() != stringToId_.size()) {
+    VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+        "idToString size {}. stringToId size {}. ",
+        idToString_.size(),
+        stringToId_.size());
+  }
+
+  if (idToString_.size() != lastId_) {
+    VELOX_STRINGIDMAP_ERROR_LOG << fmt::format(
+        "idToString {} element(s) were erased. ", lastId_ - idToString_.size());
```

Comment:
Could lastId_ - idToString_ be negative ? Similarly can stringToId_.size() != lastId_ be negative ?

Reply:
No, `last_id_` is a uint and it is always incremented to find a new entry, even elements have already been erased. It could lead to incorrect results if this rolls over the uint64 range and starts over again.

This check would becomes false immediately if `lastId_` rolls over and starts at 0 again.

Maybe this is not a good check after all because it is likely elements are erased but they don't cause integrity issues.